cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -149,6 +168,24 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	if opts.Private {
 		options = append(options, "private")
 	}
+	if opts.Overlayfs {
+		if len(opts.LowerDirs) <= 0 || len(opts.UpperDir) <= 0 || len(opts.WorkDir) <= 0 {
+			return fmt.Errorf("cannot mount %q at %q: missing arguments for overlayfs mount. lowerdir, upperdir, workdir are needed.", what, where)
+		}
+
+		var lowerDirs strings.Builder
+		for i, d := range opts.LowerDirs {
+			if i != 0 {
+				lowerDirs.WriteRune(':')
+			}
+			ed := strings.ReplaceAll(d, ",", "\\\\,")
```

Comment:
Nitpick, can you use `\\,` to avoid some of the quoting please.

Separately, can you add a comment to clarify why we need two `\` and not just one?

Reply:
Thanks a lot for the comments! I changed to using backticks. 

While taking a closer look, I realized that double escaping, while needed when I was invoking `systemd-mount` in the shell, wasn't actually needed here as Go's exec doesn't invoke the system shell. 

I am not sure what's the best way to test this as we wouldn't want to actually call `systemd-mount` from the unit test suite. I smoked tested it by copying `TestDoSystemdMount` in a new test and stopped mocking the `systemd-mount` command. Then I ran the test as root and verified that I could mount an overlay using a lowerdir that included a comma while using single escaping.